docs: Fix incorrect on-premise refs from PR #1023

alexmalins opened the pull request:
#1023 had a small number of mistakes, the most import of which broke the image links on the Examples page for the AWS on-premises examples

This PR corrects the issues and uses on-premises (or one instance of on-premise in the changelog) where needed

gabriel-tessier reviewed (changes requested):
- [ ] Revert changes in the CHANGELOG.md file as the changes are related to what changed in the version 0.16.

alexmalins commented:
hey @gabriel-tessier ! thanks for the review

> Revert changes in the CHANGELOG.md file as the changes are related to what changed in the version 0.16.

Are you sure this is the right thing to do?

0.16.0 made the breaking change `from diagrams.oci.connectivity import Customerpremisesequip` to `from diagrams.oci.connectivity import CustomerPremise` which is documented [here](https://github.com/mingrammer/diagrams/blob/master/CHANGELOG.md?plain=1#L51).

0.24.0 (incorrectly documented as [0.23.5](https://github.com/mingrammer/diagrams/blob/master/CHANGELOG.md?plain=1#L399) in the changelog) made the second breaking change from #1023 of `from diagrams.oci.connectivity import CustomerPremise` to `from diagrams.oci.connectivity import CustomerPremises`. However [360c6bc](https://github.com/mingrammer/diagrams/pull/1023/commits/360c6bc98ab913c45b600850f3b3d0114a257421) from #1023 incorrectly documented that change as `from diagrams.oci.connectivity import CustomerPremises` to `from diagrams.oci.connectivity import CustomerPremises` (i.e. Premises plural both both and after), which is what my pre-review state [a54a3cc](https://github.com/mingrammer/diagrams/pull/1097/commits/a54a3cc7b0b7199f946acb405095e74c82088046) fixes.


A wider issue is the changelog hasn't been well maintained since the v0.17.0 release. None of the subsequent releases are documented in the change log, other than @filipeaaoliveira 's work in #1023 - which was put in at the bottom of the changelog instead of at the more usual top position.

If you like I could try and bring the changelog up to date, by back filling the changes since v0.17.0 and reordering everything correctly?

gabriel-tessier commented:
@alexmalins 

Yep sorry, as you noticed the file is no more used and when I asked filipeaaoliveira to add the changelog in the PR review I forget that mingrammer were no more using the file.

Sorry for the confusion, so to sum things up:
- CHANGELOG.md file is no more used
- I confirm you can revert the changes without problem.

After that I will just wait for another maintainer to review and your changes will be merged in master.
Thank you again for the PR and sorry again for the confusion, it was better for me to inform about the changelog in the first review.
